### PR TITLE
Bump `name-registry`

### LINKED ISSUE #659: Bump `name-registry`
To:

- `forc 0.40.1`
- `fuel-core 0.18.1`
- `fuels-rs 0.42.0`

### PR DESCRIPTION
**Pull request by K1-R1**
## Type of change

- Improvement (refactoring, restructuring repository, cleaning tech debt, ...)

## Changes

The following changes have been made:

- Bump application to `forc 0.40.1` and `fuel-core 0.18.1`
- Bump test harness to `fuels 0.42.0`

## Related Issues

Closes #659

### CODE REVIEW

**Comment by K1-R1** [comment]
Have bumped without resolving #684 so that new work on the updated version can begin. This will be resolved in a future PR.

File: name-registry/README.md [hunk 1]
```diff
@@ -6,20 +6,20 @@
...
+        <img src="https://img.shields.io/badge/forc-v0.40.1-orange" />
     </a>
-    <a href="https://crates.io/crates/fuel-core/0.17.9" alt="fuel-core">
-        <img src="https://img.shields.io/badge/fuel--core-v0.17.9-yellow" />
+    <a href="https://crates.io/crates/fuel-core/0.18.1" alt="fuel-core">
+        <img src="https://img.shields.io/badge/fuel--core-v0.18.1-yellow" />
     </a>
-    <a href="https://crates.io/crates/fuels/0.40.0" alt="forc">
-        <img src="https://img.shields.io/badge/fuels-v0.40.0-blue" />
+    <a href="https://crates.io/crates/fuels/0.42.0" alt="forc">
```
Comment (Braqzen) [inline name-registry/README.md:15]: 42 or 43?

File: name-registry/README.md [hunk 2]
```diff
@@ -6,20 +6,20 @@
...
 
 <p align="center">
-    <a href="https://crates.io/crates/forc/0.37.0" alt="forc">
-        <img src="https://img.shields.io/badge/forc-v0.37.0-orange" />
+    <a href="https://crates.io/crates/forc/0.40.1" alt="forc">
+        <img src="https://img.shields.io/badge/forc-v0.40.1-orange" />
     </a>
-    <a href="https://crates.io/crates/fuel-core/0.17.9" alt="fuel-core">
-        <img src="https://img.shields.io/badge/fuel--core-v0.17.9-yellow" />
+    <a href="https://crates.io/crates/fuel-core/0.18.1" alt="fuel-core">
```
[the remark this answers is not in the document]
Comment (K1-R1) [reply name-registry/README.md:12]: I've kept versions consistent across bumps, and with 0.19 on the way I'd prefer to just get these out

File: name-registry/README.md (hunk 1 above)
Comment (K1-R1) [reply name-registry/README.md:15]: 42

**Pull request merged**